tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchBoostedSearchTests.java

```diff
@@ -0,0 +1,339 @@
+    }
+
+    /**
+     * Verify if boosts works fine with words proximity search. Files containing terms within specific distance from another one should be returned.
+     */
+    @Test(groups = { TestGroup.SEARCH })
+    public void testAftsQuery_proximitySearchBoost()
+    {
+        STEP("Search for files by name or proximity TEXT with higher priority for name filter");
+        String boostedQuery = "TYPE:'cm:content' AND (cm:name:" + SEARCH_TERM + "^5 OR TEXT:(" + SEARCH_TERM + " *(1) phrase)^0.1)";
+        SearchRequest searchRequest = req("afts", boostedQuery);
+        searchQueryService.expectResultsInOrder(searchRequest, testUser, fileWithTermInName.getName(), fileWithPhraseInContent.getName());
+
+        STEP("Search for files by name or proximity TEXT with higher priority for proximity TEXT filter");
+        String invertedBoost = "TYPE:'cm:content' AND (cm:name:" + SEARCH_TERM + "^0.1 OR TEXT:(" + SEARCH_TERM + " *(1) phrase)^5)";
+        searchRequest = req("afts", invertedBoost);
+        searchQueryService.expectResultsInOrder(searchRequest, testUser, fileWithPhraseInContent.getName(), fileWithTermInName.getName());
+    }
+
+    @Test(groups = { TestGroup.SEARCH })
+    public void testAftsQuery_dateRangeSearchBoost()
+    {
+        String timeFrom = creationTime.format(DateTimeFormatter.ISO_DATE_TIME);
+        String timeTo = afterCreationTime.format(DateTimeFormatter.ISO_DATE_TIME);
+
+        STEP("Search for files and folders by name or creation time range with higher priority for name filter");
+        String boostedQuery = "TYPE:('cm:content'^3 OR 'cm:folder') AND (cm:name:" + SEARCH_TERM + "^4 OR cm:created:['" + timeFrom + "' TO '" + timeTo + "']^0.1)";
+        SearchRequest searchRequest = req("afts", boostedQuery);
+        searchQueryService.expectResultsInOrder(searchRequest, testUser, fileWithTermInName.getName(), folderWithTermInName.getName(), fileWithTermInTitle.getName(), folderWithTermInTitle.getName());
+
+        STEP("Search for files and folders by name or creation time range with higher priority for creation time range filter");
+        String invertedBoost = "TYPE:('cm:content'^3 OR 'cm:folder') AND (cm:name:" + SEARCH_TERM + "^0.1 OR cm:created:['" + timeFrom + "' TO '" + timeTo + "']^4)";
+        searchRequest = req("afts", invertedBoost);
+        searchQueryService.expectResultsInOrder(searchRequest, testUser, fileWithTermInTitle.getName(), folderWithTermInTitle.getName(), fileWithTermInName.getName(), folderWithTermInName.getName());
+    }
+
+    /**
+     * Verify if boosts works fine with words range search. Files containing words from alphabetical range (from 'mountain' to 'phrase', this includes word 'other') should be returned.
+     */
+    @Test(groups = { TestGroup.SEARCH })
+    public void testAftsQuery_wordsRangeSearchBoost()
+    {
+        STEP("Search for files and folders by name or words in content from given range with higher priority for name filter");
```

Comment:
Not sure about "and folders" in this description

Reply:
Good catch! Removed copy-paste leftover